### PR TITLE
feat: translate web components part1

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Translate `web-components` from `Accounts` to `Credits`

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-marketplace/src/pages/Dashboard/MyProjects/MyProjects.constants.ts [hunk 1]
```diff
@@ -3,3 +3,5 @@ import { msg } from '@lingui/macro';
 export const DRAFT_ID = 'draft';
 export const CREATE_POST = msg`+ create post`;
 export const CREATE_POST_DISABLED_TOOLTIP = msg`You cannot make posts because either the project is still a draft or it doesn't have assigned a location.`;
+export const MY_PROJECTS_BUTTON_TEXT = '+ create project';
+export const MY_PROJECTS_EMPTY_TITLE = 'You have not created any projects yet';
```
Comment (wgardiner) [inline web-marketplace/src/pages/Dashboard/MyProjects/MyProjects.constants.ts:7]: Add translations

File: web-marketplace/src/pages/Dashboard/MyProjects/MyProjects.tsx [hunk 2]
```diff
@@ -71,6 +73,8 @@ const MyProjects = (): JSX.Element => {
         {activeAccountId && (
           <Grid item xs={12} md={6} lg={4}>
             <CreateProjectCard
+              buttonText={MY_PROJECTS_BUTTON_TEXT}
+              emptyTitle={MY_PROJECTS_EMPTY_TITLE}
```
Comment (wgardiner) [inline web-marketplace/src/pages/Dashboard/MyProjects/MyProjects.tsx:77]: Add translation

File: web-marketplace/src/components/organisms/RegistryLayout/RegistryLayout.constants.ts [hunk 3]
```diff
@@ -2,5 +2,7 @@ export const fullWidthRegExp: RegExp = /projects\/[a-z-]+/;
 export const TX_ERROR_MODAL_BUTTON = 'CLOSE WINDOW';
 export const KEPLR_CONNECT_BUTTON = 'connect wallet';
 export const KEPLR_LINK_TEXT = 'set up a Keplr wallet →';
+export const COOKIES_TOP_BANNER_ACCEPT_LABEL = 'accept';
+export const COOKIES_TOP_BANNER_REJECT_LABEL = 'Reject';
```
Comment (wgardiner) [inline web-marketplace/src/components/organisms/RegistryLayout/RegistryLayout.constants.ts:6]: Add translations

File: web-marketplace/src/components/organisms/RegistryLayout/RegistryLayout.CookiesTopBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import {
+  COOKIES_TOP_BANNER_ACCEPT_LABEL,
+  COOKIES_TOP_BANNER_REJECT_LABEL,
+} from './RegistryLayout.constants';
+
+export const RegistryLayoutCookiesTopBanner = () => {
+  return (
+    <CookiesTopBanner
+      acceptLabel={COOKIES_TOP_BANNER_ACCEPT_LABEL}
+      rejectLabel={COOKIES_TOP_BANNER_REJECT_LABEL}
```
Comment (wgardiner) [inline web-marketplace/src/components/organisms/RegistryLayout/RegistryLayout.CookiesTopBanner.tsx:16]: Add translations

File: web-marketplace/src/components/organisms/MediaSection.tsx [hunk 5]
```diff
@@ -34,6 +37,7 @@ const MediaSection: React.FC<React.PropsWithChildren<MediaSectionProps>> = ({
         items={
           items?.map((item, i) => (
             <ArticleCard
+              draftText={_(msg`Draft`)}
```
Comment (wgardiner) [inline web-marketplace/src/components/organisms/MediaSection.tsx:50]: Should this use `DRAFT_TEXT` from the constants file?

File: web-www/components/templates/Layout/Layout.CookiesTopBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  COOKIES_TOP_BANNER_REJECT_LABEL,
+  URL_PRIVACY,
+  URL_TERMS_SERVICE,
+} from './Layout.constants';
+
+export const LayoutCookiesTopBanner = () => {
+  return (
+    <CookiesTopBanner
+      acceptLabel={COOKIES_TOP_BANNER_ACCEPT_LABEL}
+      rejectLabel={COOKIES_TOP_BANNER_REJECT_LABEL}
```
Comment (wgardiner) [inline web-www/components/templates/Layout/Layout.CookiesTopBanner.tsx:17]: Add translations

File: web-www/components/templates/Layout/Layout.constants.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export const COOKIES_TOP_BANNER_ACCEPT_LABEL = 'accept';
+export const COOKIES_TOP_BANNER_REJECT_LABEL = 'Reject';
```
Comment (wgardiner) [inline web-www/components/templates/Layout/Layout.constants.ts:2]: Add translations

File: web-www/components/templates/home/Blog/Home.blog.tsx [hunk 8]
```diff
@@ -43,7 +44,11 @@ const BlogSection = ({ blogData }: Props): JSX.Element => {
         title={content?.header || ''}
         titleVariant="h1"
       >
-        <BlogPosts posts={posts} classes={{ item: styles.item }} />
+        <BlogPosts
+          posts={posts}
+          classes={{ item: styles.item }}
+          readMoreText={READ_MORE_TEXT}
```
Comment (wgardiner) [inline web-www/components/templates/home/Blog/Home.blog.tsx:50]: Add translation

File: web-www/components/templates/home/Blog/Home.constants.ts [hunk 9]
```diff
@@ -0,0 +1 @@
+export const READ_MORE_TEXT = 'read more';
```
Comment (wgardiner) [inline web-www/components/templates/home/Blog/Home.constants.ts:1]: Add translation

File: web-www/components/templates/resources/resourcesSection/resourcesSection.constants.ts [hunk 10]
```diff
@@ -0,0 +1 @@
+export const UPDATED_LABEL = 'Last Updated:';
```
Comment (wgardiner) [inline web-www/components/templates/resources/resourcesSection/resourcesSection.constants.ts:1]: Add translation

File: web-www/lib/constants/shared.constants.ts [hunk 11]
```diff
@@ -1 +1,8 @@
+import { ArticleType } from 'web-components/src/components/cards/ArticleCard/ArticleCard.types';
+
 export const DRAFT_TEXT = 'Draft';
+export const ARTICLE_CARD_BTN_TEXT_MAPPING: Record<ArticleType, string> = {
+  video: 'watch video',
+  article: 'read article',
+  podcast: 'listen to podcast',
```
Comment (wgardiner) [inline web-www/lib/constants/shared.constants.ts:7]: Add translation

**Review by wgardiner — COMMENTED** [review]
Looks like there are a number of translations that need to be added. There are also issues with the review deploy of the website (https://app.netlify.com/sites/regen-website/deploys/66d728f3b6ac500008e505ed looks like linter errors related to unsorted imports are breaking the build?? seems odd to me) and with the storybook review.

**Comment by flagrede** [comment]
@wgardiner This PR only handles translations from `web-components` (Accounts to Credits components). All `web-marketplace` raw strings should have been translated in other open PRs. Once all PRs have been merged, I'll rerun a check with the linter to see if we missed any.
Regarding `web-www,`, these translations are out of the current scope; we're focusing on the marketplace for now (/cc @blushi).

File: web-marketplace/src/pages/Dashboard/MyProjects/MyProjects.constants.ts (hunk 1 above)
Comment (flagrede) [reply web-marketplace/src/pages/Dashboard/MyProjects/MyProjects.constants.ts:7]: It was already translated in another PR.

File: web-marketplace/src/components/organisms/RegistryLayout/RegistryLayout.constants.ts (hunk 3 above)
Comment (flagrede) [reply web-marketplace/src/components/organisms/RegistryLayout/RegistryLayout.constants.ts:6]: It was already translated in another PR.

File: web-marketplace/src/components/organisms/RegistryLayout/RegistryLayout.CookiesTopBanner.tsx (hunk 4 above)
Comment (flagrede) [reply web-marketplace/src/components/organisms/RegistryLayout/RegistryLayout.CookiesTopBanner.tsx:16]: Same.

File: web-marketplace/src/components/organisms/MediaSection.tsx (hunk 5 above)
Comment (flagrede) [reply web-marketplace/src/components/organisms/MediaSection.tsx:50]: Same.

File: web-www/components/templates/Layout/Layout.CookiesTopBanner.tsx (hunk 6 above)
Comment (flagrede) [reply web-www/components/templates/Layout/Layout.CookiesTopBanner.tsx:17]: Out of scope.

File: web-www/components/templates/home/Blog/Home.blog.tsx (hunk 8 above)
Comment (flagrede) [reply web-www/components/templates/home/Blog/Home.blog.tsx:50]: Same.

File: web-www/lib/constants/shared.constants.ts (hunk 11 above)
Comment (flagrede) [reply web-www/lib/constants/shared.constants.ts:7]: Same.

**Comment by wgardiner** [comment]
Ah okay, I thought that might be the case. Looks good!

**Review by wgardiner — APPROVED** [review]

**Comment by blushi** [comment]
image
 
image

Some translations missing.

**Comment by blushi** [comment]
> Ah okay, I thought that might be the case. Looks good!

Indeed, translating the website is out of scope so that means that lingui is only available in web-marketplace, so we need to provide any displayed strings as props to web-components.

**Pull request merged**